tf/main.tf

```diff
@@ -0,0 +1,35 @@
+provider "google" {
+  project = var.project
+  region  = "us-east1"
+  version = "3.25"
+}
+
+provider "google-beta" {
```

Comment:
do you need google-beta?

Reply:
I dunno, @mjvaldez did this work.

Reply:
it's not a big deal, but everything here looks pretty vanilla so I'd be surprised if it's actually necessary.

If we pull in the beta provider, it's probably a good practice to document why it's needed, so we have an obvious path to removing it when it's no longer needed.

Reply:
Any thoughts, @mjvaldez ?

Reply:
@mjvaldez

Reply:
This is the @mjvaldez mega thread

Reply:
This just configured the beta provider but it's never used. This part can go.

Reply:
This configures the beta provider in case modules we call use it. I'm not sure if it's getting used now but I'd just consider it boilerplate.

Yeah some comments wouldn't hurt. I think I'd like to move this stuff into the rhino repo at some point anyway..

Reply:
Getting in the habit of always pulling in the beta provider as a form of boilerplate, regardless of whether we need it seems like it might introduce stability problems, or at the very least, confusion about whether we're using generally available or beta GCP products. My preference is to omit it unless it's clear that we need it for some reason, then have a plan to deprecate the beta usage once the feature is generally available.

I was just reading through the TF docs, and it seems like unless we specify using `google-beta` as a provider in a resource block, Terraform defaults to using the `google` provider for that block ([source](https://www.terraform.io/docs/providers/google/guides/provider_versions.html#using-the-google-beta-provider)). So I think in our case, since we aren't specifying a `google-beta` provider in any of our resource blocks here, it's actually not being used at all. Based off of that, my preference in this case is to remove the `google-beta` provider definition here since its presence is confusingly signaling that it might be in use, when in fact it's not.

Reply:
I agree this sucks but it's not the resources defined directly in this configuration that need it, it's actually modules that the configuration uses. That said, this configuration doesn't use any modules, so it can go (sorry didn't notice that earlier).

There are a couple of ways to make this less awful and it's really an artifact of roughing in the initial implementation. I'll make sure it goes away soon.

Reply:
ahhhh, I see. You mentioned modules, but I was thinking resources. Like you say, there's no modules in use here, but I'm glad to understand the distinction because, for instance, I'll need to use google-beta in the counter sql TF since I'll be invoking our `mysql` module https://gitlab.com/plos/gcp-global/-/blob/master/modules/cloudsql/main.tf#L17.

@egh, it sounds like the upshot here is that we should just delete this `google-beta` provider here since it's not needed in this case. But if it were needed, we should add a comment explaining why.

Reply:
I'll delete it.

Reply:
5eccfeb